force-app/main/default/classes/CourseRegistrationHandler.cls

```diff
@@ -0,0 +1,135 @@
+global class CourseRegistrationHandler extends MyTriggers {
+    global override void onAfterDelete() {
+        getCourseIdToReduce((List<CourseRegistration__c>) records);
+    }
+
+    global override void onAfterUndelete() {
+        getCourseIdToIncrease((List<CourseRegistration__c>) records);
+    }
+
+    global override void onAfterInsert() {
+        getCourseIdToIncrease((List<CourseRegistration__c>) records);
+    }
+
+    global override void onAfterUpdate(Map<Id, sObject> triggerOldMap) {
+        List<String> fieldsToCheck = new List<String>{'Status__c'};
+        Id courseId;
+        for (CourseRegistration__c cr : (List<CourseRegistration__c>) records) {
+            if(cr.Status__c == 'Påmeldt' && 
+               MyTriggers.hasChangedFields(fieldsToCheck, cr, triggerOldMap.get(cr.Id))) 
+               {
```

Comment:
Prøv å endre alle if-er og else if-er til å bare vurdere boolean, og lag alle booleans på toppen. Det gjør koden mer lesbar, og det blir enklere å forstå hva hver if gjør.

```suggestion
           Boolean statusIsUpdated = MyTriggers.hasChangedFields(fieldsToCheck, cr, triggerOldMap.get(cr.Id)); // denne bruker du i hver if, og det trengs ikke. Gjør det bare vanskeligere å lese.
 
            if (statusIsUpdated) continue; // skips this record, because no change to the status

            CourseRegistration__c oldCr = (CourseRegistration__c) triggerOldMap.get(cr.Id);
            Boolean newIsAttending = cr.Status__c == 'Påmeldt';
            Boolean newIsNotAttending = cr.Status__c == 'Avmeldt';
            Boolean newIsWaitListed = cr.Status__c == 'Venteliste';

            Boolean oldIsAttending = oldCr.Status__c == 'Påmeldt';
            Boolean oldIsNotAttending = oldCr.Status__c == 'Avmeldt';
            Boolean oldIsWaitListed = oldCr.Status__c == 'Venteliste';

            Boolean wasWaitListed_NowAttending = oldIsWaitListed && newIsAttending;
            Boolean wasAttending_NowNotAttending = oldIsAttending && newIsNotAttending;
            Boolean wasAttending_NowWaitListed = oldIsAttending && newIsWaitListed;
            // .... 
            // legg til resten i sammen format
            
            if ( newIsAttending ) {
```